recipe/build.sh

```diff
@@ -1,46 +1,97 @@
-cat rts.pkg
-ghc-pkg update rts.pkg
-echo "Setting x86_64-conda_cos6-linux-gnu-gcc"
-echo "GCC version"
-x86_64-conda_cos6-linux-gnu-gcc --version
-echo "CC version"
-x86_64-conda_cos6-linux-gnu-cc --version
-echo "Collect2"
-x86_64-conda_cos6-linux-gnu-cc -print-prog-name=collect2
-echo "ld"
-x86_64-conda_cos6-linux-gnu-cc -print-prog-name=ld
-echo "LD version"
-x86_64-conda_cos6-linux-gnu-ld --version
-echo "LD help"
-x86_64-conda_cos6-linux-gnu-ld --help
-export EXTRA_CONFIGURE_OPTS=" --extra-include-dirs=$PREFIX/include --extra-lib-dirs=$PREFIX/lib ";
-sed -i -- 's/collect2 //g' cabal-install/bootstrap.sh
-
-ghc-pkg recache
-cd cabal-install
-echo "Extra configure opts"
-echo "$EXTRA_CONFIGURE_OPTS"
-sed -i -- 's/export LD=$LINK/export LINK=x86_64-conda_cos6-linux-gnu-cc/g' bootstrap.sh
-
-sed -i -- 's/${GHC} --make ${JOBS} ${PKG_DBS} Setup -o Setup/${GHC} -lgmp -threaded -pgmc x86_64-conda_cos6-linux-gnu-cc -pgml x86_64-conda_cos6-linux-gnu-cc --make ${JOBS} ${PKG_DBS} Setup -o Setup/g' bootstrap.sh
-cat bootstrap.sh
-export GHC=`which ghc`
-strings $GHC
-./bootstrap.sh --no-doc
+#!/usr/bin/env bash
+
+set -eux
+
+# Environment setup
+export PKG_CONFIG_PATH="${PREFIX}/lib/pkgconfig:${PKG_CONFIG_PATH:+:}${PKG_CONFIG_PATH:-}"
+export LD_LIBRARY_PATH="${PREFIX}/lib:${BUILD_PREFIX}/lib${LD_LIBRARY_PATH:+:}${LD_LIBRARY_PATH:-}"
+export LIBRARY_PATH="${PREFIX}/lib:${BUILD_PREFIX}/lib${LIBRARY_PATH:+:}${LIBRARY_PATH:-}"
+export PATH="${BUILD_PREFIX}/ghc-bootstrap/bin${PATH:+:}${PATH:-}"
+
+unset build_alias
+unset host_alias
+
+# Clean cabal environment
+clean_cabal() {
+  eval ${CABAL} clean
+  rm -rf dist-newstyle
+  rm -rf ~/.cabal/store ~/.cabal/packages
+  eval ${CABAL} update
+}
+
+# Install cabal with given parameters
+install_cabal() {
+  local install_dir="${1}"
+  eval ${CABAL} install \
+    --project-file=cabal.release.constraints.project \
+    --installdir="${install_dir}" \
+    --install-method=copy \
+    --minimize-conflict-set \
+    ${CABAL_CONFIG_FLAGS:-} \
+    cabal-install
+}
+
+# Main build process
+main() {
+  # Initialize package database
+  ghc-pkg recache
+  
+  # Configure GHC for Windows compatibility
+  if [[ "${target_platform}" == win-* ]]; then
+    export CC=${GCC}
+    export CABAL_CONFIG_FLAGS="--enable-static --disable-shared --ghc-options=-static"
+  elif [[ "${target_platform}" == osx-* ]]; then
+    settings_file=$(find "${BUILD_PREFIX}"/ghc-bootstrap/lib -name settings)
+    cat "${settings_file}"
+    # perl -i -pe 's#("C compiler link flags", ")([^"]*)"#\1\2 -L\$topdir/../../../../lib"#g' "${settings_file}"
+    # perl -i -pe 's#("C compiler link flags", ")([^"]*)"#\1\2 -Wl,-rpath,@loader_path/../lib"#g' "${settings_file}"
+    export CABAL_CONFIG_FLAGS="-v1 --enable-static --disable-shared --ghc-options=-optl-Wl,-dead_strip"
+  else
+    export CABAL_CONFIG_FLAGS=""
+  fi
+
+  # Install bootstrapping cabal from conda-forge
+  if ! conda create -n cabal_env -y -c conda-forge cabal; then
```

Comment:
Personally, I think it is fine to use the pre-built cabal from haskell.org every time. We do this also for go or bazel.

Reply:
I removed the conda bootstrapping